src/Lib/ClientOptionsValidator.php

```diff
@@ -0,0 +1,171 @@
+     *              - mode                  string  'test' or 'live'. Default: 'live'
+     *              - logger                Psr\Log\LoggerInterface The logger instance to use for errors/warnings
+     *              - user_agent_component  array[user_agent => version] user_agent and version should be string
+     *
+     * @throws ParamsError
+     */
+    public static function validateOptions(array $options = array())
+    {
+        $config = [
+            'api_root' => [
+                Client::TEST_MODE => Client::SANDBOX_API_URL,
+                Client::LIVE_MODE => Client::LIVE_API_URL
+            ],
+            'force_tls' => 2,
+            'mode' => Client::LIVE_MODE,
+            'logger' => new NullLogger(),
+        ];
+
+        if (isset($options['api_root'])) {
+            $config['api_root'] = self::validateApiRootOption($options['api_root']);
+        }
+
+        if (isset($options['force_tls'])) {
+            $config['force_tls'] = self::validateForceTLSOption($options['force_tls']);
+        }
+
+        if (isset($options['mode'])) {
+            $config['mode'] = self::validateModeOption($options['mode']);
+        }
+
+        if (isset($options['logger'])) {
+            $config['logger'] = self::validateLoggerOption($options['logger']);
+        }
+
+        if (isset($options['user_agent_component'])) {
+            $config['user_agent_component'] = self::validateUserAgentComponentOption($options['user_agent_component']);
+        }
+        return $config;
+    }
+
+    /**
+     * @param $api_root
+     *
+     * @return array|string[]
+     * @throws ParamsError
+     */
+    public static function validateApiRootOption($api_root)
+    {
+        if (is_string($api_root)) {
+            return [
+                Client::TEST_MODE => $api_root,
+                Client::LIVE_MODE => $api_root
+            ];
+        }
+        if (isset($api_root[Client::TEST_MODE]) && isset($api_root[Client::LIVE_MODE])) {
+            return [
+                Client::TEST_MODE => $api_root[Client::TEST_MODE],
+                Client::LIVE_MODE => $api_root[Client::LIVE_MODE]
+            ];
+        }
+        throw new ParamsError('option \'api_root\' is not configured properly');
+    }
+
+    /**
+     * @param bool|int $force_tls
+     *
+     * @return bool|int
+     * @throws ParamsError
+     */
+    public static function validateForceTLSOption($force_tls)
+    {
+        if ($force_tls === true || $force_tls === 2) {
+            return 2;
+        }
+        if (in_array($force_tls, [0, 1, false])) {
+            return $force_tls;
+        }
+        throw new ParamsError('option \'force_tls\' is not configured properly');
+    }
+
+    /**
+     * @param $mode
+     *
+     * @return mixed
+     * @throws ParamsError
+     */
+    public static function validateModeOption($mode)
+    {
+        if (in_array($mode, [Client::TEST_MODE, $mode === Client::LIVE_MODE])) {
```

Comment:
if `$mode === 'live'`, we will test if that is in array `['test', true]` 

I expect that will not works find ... 

There is something to fix here and a tests to add 😉

Reply:
Thanks for catching this one !